src/test/java/org/folio/rest/api/AuthorityStorageTest.java

```diff
@@ -106,6 +109,10 @@ public void deleteByIdNotFound() {
   public void putById() {
     assertEquals(0, authoritiesClient.getAll().size());
     createAuthRecords(1);
+
+    // Clear events to help avoid delayed message problems.
+    removeAllEvents();
```

Comment:
Why is this being done part way through a write operation? Is it to remove all captured events published during the write because they aren't relevant?

Reply:
The `createAuthRecords(1);` results in a CREATE message in the Kafka logs.
For whatever reason that CREATE sometimes appears after the UPDATE.

This is an attempt to help avoid these delayed message problems.
Emphasis on attempt.

The real solution is to not rely on the messaging order as we currently do but that is a far bigger task and out of scope of my goals here.

Originally, the `removeAllEvents()` called `commit()` here to help force Kafka to impose some influence over the order. This actually revealed the problem with commit/autocommit in Kafka as a major factor in the random out of order messages.

I removed the commit() and disabled autocommit but left this particular call in places.
The areas in code that have the most consistent random failures based on CREATE appearing after UPDATE retain this function call in the middle of tests and usually after a create operation. I believe I also added that short comment very briefly describing the purpose of the call.

Reply:
> The `createAuthRecords(1);` results in a CREATE message in the Kafka logs. For whatever reason that CREATE sometimes appears after the UPDATE.

By Kafka logs, do you mean the logs produced by Kafka, the topic itself or the fake consumer that collects in-memory lists?

> This is an attempt to help avoid these delayed message problems. Emphasis on attempt. The real solution is to not rely on the messaging order as we currently do but that is a far bigger task and out of scope of my goals here.

Agreed.

> The areas in code that have the most consistent random failures based on CREATE appearing after UPDATE retain this function call in the middle of tests and usually after a create operation. I believe I also added that short comment very briefly describing the purpose of the call.

Cool. The reason I asked for more info is because the comments often say what removeAllEvents would do and would refer to the `delayed messages problem` however not what that was in the particular context.

Reply:
> By Kafka logs, do you mean the logs produced by Kafka, the topic itself or the fake consumer that collects in-memory lists?

Sorry, I mis-typed.
I was thinking of the messages kafka manages/sends/receives rather than a "log".

> Cool. The reason I asked for more info is because the comments often say what removeAllEvents would do and would refer to the delayed messages problem however not what that was in the particular context.

I can add more verbose comments in these cases then.